config/routes.rb

```diff
@@ -263,6 +263,9 @@
     resources :reparents, only: [:new, :create]
   end
 
+  # only admin should be able to access it.. right ?
```

Comment:
Let's allow any moderator to use it. It'd be good to give them insight into the kinds of partial outages that people might ask about when jobs fail.

Reply:
Right now, I have configured `basic_http_authentication` with credentials stored in Rails credentials. Integrating with our native authentication workflow may require some more work.

Reply:
I found [a tutorial](https://kyrylo.org/rails/2024/08/07/how-to-add-mission-control-to-your-rails-app-and-protect-it-with-basic-auth.html) for basic auth. The interesting thing was setting `config.mission_control.jobs.base_controller_class` - if you set that to our `Mod::ModeratorController` (from `app/controllers/mod/moderator_controller.rb`, be careful, there are several controllers with very similar names) would it use our authentication?

Reply:
That would be very convenient for the moderators. But, it will require more
work. For example, ModController inherits from ApplicationController which
has n+1 query check middleware, that will cause problem with
mission-control gem.

My guess is if we make a new JobModcontroller class inheriting from
Actioncontroller::Base, maybe we can make it work.

Reply:
Something like this is working. 
https://github.com/rahul1990gupta/lobsters/pull/2

Reply:
another option is to extract the authentication module into a concern and use it for `ApplicationController` and `JobsModController`
https://github.com/rahul1990gupta/lobsters/pull/3

Reply:
A Concern sounds like the exact right tool for the job, please extract one to share authentication between the mission control and mod controllers. Thanks for figuring out our options here.